clients/js/genjs.go

```diff
@@ -338,10 +374,12 @@ var methodTmplStr = `
 
           switch (response.statusCode) {
             {{ range $response := .Responses }}case {{ $response.StatusCode }}:{{if $response.IsError }}
+			  var err = new Errors.{{ $response.Name }}(body || {});
+			  responseLog(this.logger, response, err);
```

Comment:
@cozmo seems like I'm having issues with `this` here. Would love to walk through with you tomorrow?